File: Source/MKL/examples/vmlf/source/vsunpackv.f
	programm MKL_VML_TEST

	include "func_interv.fi"

	real*4 sA(VEC_LEN)
	real*4 sB1(VEC_LEN)
	real*4 sB2(VEC_LEN)
	integer i
	integer ia(VEC_LEN)

	i=0

	do i=1,VEC_LEN
		sA(i)=i/1.0
		sB1(i)=0.0
		sB2(i)=0.0
		ia(i)=vec_len-i
	end do

	call VSPACKV(vec_len,sA,ia,sB1)
	call VSUNPACKV(vec_len,sB1,sB2,ia)

	print *,"vsUnpackV test/example programm"
	print *,""
	print *,""

	print *,"           Before packing             After packing          After Unpacking"
	print *,"==============================================================================="
	do i=1,VEC_LEN
		print 10,sA(i),sB1(i),sB2(i)
	end do

10	format (F25.13,F25.13,F25.13)

	end

File: Source/MKL/examples/vmlc/source/vsunpackv.c
#include <stdio.h>
#include <math.h>
#include "func_interv.h"

#include "mkl_vml.h"

int main()
{
  float fA[VEC_LEN],fB1[VEC_LEN],fB2[VEC_LEN];
  int ia[VEC_LEN];

  int i=0,vec_len=VEC_LEN;

  for(i=0;i<vec_len;i++) {
    fA[i]=(float)i+1;
    fB1[i]=0.0;
    fB2[i]=0.0;
    ia[i]=vec_len-i-1;
  }

  vsPackV(vec_len,fA,ia,fB1);
#if defined(MKL_VML_MIXED)
  vsUnpackV(vec_len,fB1,fB2,ia);
#elif defined(MKL_VML_LOWER)
  vsunpackv(&vec_len,fB1,fB2,ia);
#elif defined(MKL_VML_UPPER)
  VSUNPACKV(&vec_len,fB1,fB2,ia);
#else
# error You must define MKL_VML_CDECL for CDECL call convention or MKL_VML_STDCALL for STDCALL call convention.
#endif

  printf("vsUnpackV test/example programm\n\n");
  printf("           Before packing             After packing          After Unpacking\n");
  printf("===============================================================================\n");
  for(i=0;i<VEC_LEN;i++) {
    printf("% 25.14f % 25.14f % 25.14f\n",fA[i],fB1[i],fB2[i]);
  }
}
